ygen/ir.go

```diff
@@ -158,11 +197,30 @@ type NodeDetails struct {
 	// the output code, using the output of the language-specific
 	// type mapping provided by calling the LangMapper interface.
 	LangType *MappedType
-	// MapPaths describes the paths that the output node should
+	// MappedPaths describes the paths that the output node should
 	// be mapped to in the output code - these annotations can be
 	// used to annotation the output code with the field(s) that it
 	// corresponds to in the YANG schema.
-	MapPaths [][]string
+	MappedPaths [][]string
+	// MappedModules describes the path elements' belonging modules that
+	// the output node should be mapped to in the output code - these
+	// annotations can be used to annotation the output code with the
+	// field(s) that it corresponds to in the YANG schema.
+	MappedModules [][]string
+	// ShadowMappedPaths describes the shadow paths (if any) that the output
+	// node should be mapped to in the output code - these annotations can
+	// be used to annotation the output code with the field(s) that it
+	// corresponds to in the YANG schema.
+	// Shadow paths are paths that have sibling config/state values
+	// that have been compressed out due to path compression.
+	ShadowMappedPaths [][]string
+	// ShadowMappedModules describes the shadow path elements' belonging
+	// modules (if any) that the output node should be mapped to in the
+	// output code - these annotations can be used to annotation the output
+	// code with the field(s) that it corresponds to in the YANG schema.
+	// Shadow paths are paths that have sibling config/state values
+	// that have been compressed out due to path compression.
+	ShadowMappedModules [][]string
```

Comment:
Is there any case in which `ShadowMappedModules` is not equal to `MappedModules`? It seems like there should not be such a case because it would mean that we were doing compression across modules for OpenConfig (which wouldn't be allowable), but I guess this theoretically can happen through augmenting in a `state` leaf for a `config` leaf and with compression enabled?

Reply:
You mentioned one way. Another way is if the `ShadowMappedPaths` is different then the dimensions of the `ShadowMappedModules` will match that of `ShadowMappedPaths`. I thought keeping them independent also keeps this simpler to understand.

Reply:
Sounds good to me. I was just wanting to check that we were on the same page here.